tests/src/test/java/alluxio/underfs/UnderStorageSystemInterfaceIntegrationTest.java

```diff
@@ -195,6 +238,46 @@ public void listTest() throws IOException {
   }
 
   /**
+   * Tests if list correctly returns file or folder names for a large directory.
+   */
+  @Test
+  public void listLargeDirectoryTest() throws IOException {
+    String topLevelDirectory = PathUtils.concatPath(mUnderfsAddress, "testDirNonEmpty1");
+
+    final String filePrefix = "a_";
+    final String folderPrefix = "b_";
+
+    final int numFiles = 1500;
+    final int numFolders = numFiles;
+
+    // Make top level directory
+    mUfs.mkdirs(topLevelDirectory, false);
+
+    // Make the children files
+    for (int i = 0; i < numFiles; ++i) {
+      createEmptyFile(PathUtils.concatPath(topLevelDirectory, filePrefix
+          + String.format("%04d", i)));
+    }
+    // Make the children folders
+    for (int i = 0; i < numFolders; ++i) {
```

Comment:
These two tests share quite some amount of code. Shall we extract them into a `prepare` method?